lib/fog/kubevirt/compute/compute.rb

```diff
@@ -339,18 +327,51 @@ def create_client(path, version)
             :port   => @port,
             :path   => path
           )
-          client = Kubeclient::Client.new(
-            url.to_s,
-            version,
-            @opts
-          )
+
+          client = if @kubevirt_token.to_s.empty?
+            create_client_from_config(url, version)
+          else
+            create_client_from_token(url, version)
+          end
+
           wrapped_client = ExceptionWrapper.new(client)
           @clients[key] = wrapped_client
 
           # Return the client:
           wrapped_client
         end
 
+        def create_client_from_token(url, version)
+          # Prepare the TLS and authentication options that will be used for the standard Kubernetes API
+          # and also for the KubeVirt extension:
+          @opts = {
+            :ssl_options  => {
+              :verify_ssl => OpenSSL::SSL::VERIFY_NONE,
+            },
+            :auth_options => {
+              :bearer_token => @kubevirt_token
+            }
+          }
+
+          Kubeclient::Client.new(
+            url.to_s,
+            version,
+            @opts
+          )
+        end
+
+        def create_client_from_config(url, version)
+          config = Kubeclient::Config.read(ENV['KUBECONFIG'] || ENV['HOME']+'/.kube/config')
```

Comment:
Will this produce a useful error in case none of the above provided ? (neither env variable or a file)

Reply:
Yes. it will complain (raise error) about missing configuration fail.